components/clarinet-cli/tests/cli.rs

```diff
@@ -0,0 +1,124 @@
+use std::{fs, path::Path, process::Command};
+
+use clarinet_files::{FileLocation, ProjectManifest, ProjectManifestFile};
+
+#[track_caller]
+fn parse_manifest(project_dir: &Path) -> ProjectManifest {
+    let manifest_path = project_dir.join("Clarinet.toml");
+    let manifest_str = fs::read_to_string(&manifest_path).expect("Failed to read Clarinet.toml");
+    let manifest_file: ProjectManifestFile = toml::from_str(&manifest_str).unwrap();
+    let location = FileLocation::from_path(manifest_path);
+    ProjectManifest::from_project_manifest_file(manifest_file, &location, false).unwrap()
+}
+
+#[track_caller]
+fn create_new_project(project_name: &str) -> tempfile::TempDir {
+    let temp_dir = tempfile::tempdir().expect("Failed to create temp dir");
+    let status = Command::new(env!("CARGO_BIN_EXE_clarinet"))
+        .args(["new", project_name, "--disable-telemetry"])
+        .current_dir(&temp_dir)
+        .status();
+    assert!(status.unwrap().success());
+    temp_dir
+}
+
+#[test]
+fn test_new_project() {
+    let project_name = "test_project";
+    let temp_dir = tempfile::tempdir().expect("Failed to create temp dir");
+    let cmd = Command::new(env!("CARGO_BIN_EXE_clarinet"))
+        .args(["new", project_name, "--disable-telemetry"])
+        .current_dir(&temp_dir)
+        .output()
+        .expect("Failed to execute clarinet new");
+    assert!(cmd.status.success());
+
+    let expected_lines = [
+        "Telemetry disabled. Clarinet will not collect any data on this project.",
+        "Created directory test_project",
+        "Created directory contracts",
+        "Created directory settings",
+        "Created directory tests",
+        "Created file Clarinet.toml",
+        "Created file settings/Mainnet.toml",
+        "Created file settings/Testnet.toml",
+        "Created file settings/Devnet.toml",
+        "Created directory .vscode",
+        "Created file .vscode/settings.json",
+        "Created file .vscode/tasks.json",
+        "Created file .gitignore",
+        "Created file .gitattributes",
+        "Created file package.json",
+        "Created file tsconfig.json",
+        "Created file vitest.config.js",
+    ];
+    let stdout = String::from_utf8_lossy(&cmd.stdout);
+    let stdout_lines: Vec<_> = stdout.lines().map(str::trim).collect();
+    let actual = stdout_lines[..expected_lines.len()].to_vec();
```

Comment:
No need to allocate a `Vec`:

```suggestion
    let actual = &stdout_lines[..expected_lines.len()];
```